core/shared/platform/windows/win_socket.c

```diff
@@ -124,12 +157,25 @@ int
 os_socket_accept(bh_socket_t server_sock, bh_socket_t *sock, void *addr,
                  unsigned int *addrlen)
 {
+    CHECK_VALID_SOCKET_HANDLE(server_sock);
+
     struct sockaddr addr_tmp;
     unsigned int len = sizeof(struct sockaddr);
 
-    *sock = accept(server_sock, (struct sockaddr *)&addr_tmp, &len);
+    *sock = wasm_runtime_malloc(sizeof(windows_handle));
+
+    if (*sock == NULL) {
+        errno = ENOMEM;
+        return BHT_ERROR;
+    }
+
+    (*sock)->type = windows_handle_type_socket;
+    (*sock)->access_mode = windows_access_mode_read_only;
```

Comment:
why is that read only? Shouldn't we also be able to write to the socket? E.g. to send data to the client.

Reply:
yup, not sure why I put read only there.